### PR TITLE
Add support for new loadgen API

### PR DESCRIPTION
**Pull request by bboston7**
This PR adds support for the new distribution parameters added in https://github.com/stellar/stellar-core/pull/4248.  It is #155 with the mixed soroban/classic max TPS mission removed.

I tested these changes with the following missions using an image built from the linked stellar-core PR:
* `SorobanInvokeHostLoad`
* `SorobanConfigUpgrades`
* `SimplePayment`
* `LoadGeneration`
* `LoadGenerationWithTxSetLimit`
* `ProtocolUpgradeWithLoad`

### CODE REVIEW

File: src/FSLibrary/StellarCoreHTTP.fs
Comment (marta-lokhova) [inline src/FSLibrary/StellarCoreHTTP.fs:275]: Why remove this? Aren't some missions using these defaults? (seems a bit easier to just use a pre-set rather than configure invoke host options manually, which is pretty involved)

File: src/FSLibrary/StellarCoreHTTP.fs
Comment (marta-lokhova) [inline src/FSLibrary/StellarCoreHTTP.fs:182]: These configs are part of loadgen configs, why do we remove them here?

File: src/FSLibrary/MissionHistoryGenerateAndCatchup.fs
Comment (marta-lokhova) [inline src/FSLibrary/MissionHistoryGenerateAndCatchup.fs:18]: Isn't this a no-op now, since we don't set number of accounts and txs anymore? I think we can keep all the places that use the defaults (withNominalLoad, basic presets for account generation, pay, etc) as-is as to not accidentally change the existing missions.

File: src/FSLibrary/StellarCoreHTTP.fs
Comment (bboston7) [reply src/FSLibrary/StellarCoreHTTP.fs:182]: These configs aren't HTTP options anymore. All of these became distributions rather than low/high values, and in doing so I moved them to the config file instead of using HTTP options. They're now declared as config file options in `Program.fs`

File: src/FSLibrary/StellarCoreHTTP.fs
Comment (bboston7) [reply src/FSLibrary/StellarCoreHTTP.fs:275]: My other comment mostly explains this, but because these aren't HTTP parameters anymore they can't be defined here. The default is now the builtin loadgen defaults (by setting empty distributions). Missions that want easy-to-use presets can use `WithSmallLoadgenOptions` and `WithMediumLoadgenOptions` which are set to respect the network limits used in many of these tests. This change updates all of the soroban loadgen missions to use one of these presets based on the limits in the mission.

File: src/FSLibrary/MissionHistoryGenerateAndCatchup.fs
Comment (bboston7) [reply src/FSLibrary/MissionHistoryGenerateAndCatchup.fs:18]: I don't think so? My reading of the documentation for [the `with` keyword for record updates](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/copy-and-update-record-expressions) (which is how `WithNominalLoad` and `WithSmallLoadgenOptions` work) is that it only updates the specified fields and leaves the others untouched. So `WithNominalLoad` would return a new context with some fields updated, and pass that to `WithSmallLoadgenOptions` which will update different fields. So the end effect of this is that `context` has both the `WithNominalLoad` AND `WithSmallLoadgenOptions` defaults (there's no overlap between these updates).

I'm I misunderstanding how record update in F# works? I'll see if I can test a simple example in the REPL to double check.

Comment (bboston7) [reply src/FSLibrary/MissionHistoryGenerateAndCatchup.fs:18]: I was able to check this with a simple example in the REPL:

```F#
> type Rec = { X : int ; Y : int ; Z : int };;
type Rec =
  { X: int
    Y: int
    Z: int }

> type Rec with
-     member self.UpdateX : Rec = { self with X = 4 }
-     member self.UpdateY : Rec = { self with Y = 5 };;
type Rec with
  member UpdateX : Rec
type Rec with
  member UpdateY : Rec

> let r = { Rec.X = 1 ; Rec.Y = 2 ; Rec.Z = 3 };;
val r : Rec = { X = 1
                Y = 2
                Z = 3 }

> r.UpdateX.UpdateY
- ;;
val it : Rec = { X = 4
                 Y = 5
                 Z = 3 }
```

So `context` should have both sets of updates, not just `WithSmallLoadgenOptions`.

**Comment by bboston7** [comment]
I rebased this on top of `main`

File: src/FSLibrary/StellarCoreHTTP.fs
Comment (marta-lokhova) [reply src/FSLibrary/StellarCoreHTTP.fs:182]: Ah yes, makes sense. Sorry, I think I reviewed this change before ☕  :)

**Review by marta-lokhova — APPROVED** [review]
LGTM! Will merge this PR once the corresponding stellar-core PR is landed.

**Comment by marta-lokhova** [comment]
https://github.com/stellar/stellar-core/pull/4248 merged, so I think we're good to go!

**Pull request merged**
